dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/InternalLogger.kt

```diff
@@ -96,6 +96,16 @@ interface InternalLogger {
         additionalProperties: Map<String, Any?>? = null
     )
 
+    /**
+     * Logs a specific metric from the internal implementation. The metric will leverage the new
+     * added additionalProperties in the
+     * [com.datadog.android.telemetry.model.TelemetryDebugEvent.Telemetry] and will be sent as
+     * [com.datadog.android.telemetry.model.TelemetryDebugEvent].
+     * @param messageBuilder the lambda building the metric message
+     * @param additionalProperties additional properties to add to the metric
+     */
+    fun metric(messageBuilder: () -> String, additionalProperties: Map<String, Any?>)
```

Comment:
Function name should be action verbs. We should consider `logMetric`. Alternatively, we could instead use an `InternalLogger.Level.METRIC` instead, and reuse the existing top level methods. WDYT?

Reply:
i'd rather keep a new method here to match what `iOS` added in the `InternalLogger` but will change it to `logMetric`